File: touch_bar_macos/macos/Classes/Models/TouchBarLabel.swift
// Copyright (c) 2020 Eduardo Vital Alencar Cunha
// Use of this source code is governed by a MIT-style license that can be
// found in the LICENSE file.

import Foundation

class TouchBarLabel: NSCustomTouchBarItem, TouchBarItem {
  required init?(identifier: NSTouchBarItem.Identifier, withData itemData: NSDictionary) {
    super.init(identifier: identifier)

    // extract item data
    guard let label = itemData["label"] as? String else { return nil }
    let accessibilityLabel = itemData["accessibilityLabel"] as? String
    let textColor = itemData["color"] as? NSColor

    // setup NSTextField with extracted data
    let textField = NSTextField(labelWithString: label)
    textField.setAccessibilityLabel(accessibilityLabel)
    textField.textColor = textColor

    self.view = textField
  }

  func update(data: NSDictionary) {
    if let label = data["label"] as? String {
      (self.view as! NSTextField).stringValue = label
    } else if let accessibilityLabel = data["accessibilityLabel"] as? String {
      (self.view as! NSTextField).setAccessibilityLabel(accessibilityLabel)
    } else if let textColor = data["color"] as? NSColor {
      (self.view as! NSTextField).textColor = textColor
    }
  }

  required init?(coder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }
}

File: touch_bar_platform_interface/lib/models/touch_bar_items/touch_bar_label.dart
// Copyright (c) 2020 Eduardo Vital Alencar Cunha
// Use of this source code is governed by a MIT-style license that can be
// found in the LICENSE file.

part of 'package:touch_bar_platform_interface/models/touch_bar_items/mixins/labelable_item.dart';

/// A touch bar item with a text with a single style.
class TouchBarLabel extends TouchBarItem with LabelableItem {
  /// Creates a new label item with the given [identifier] and [label].
  TouchBarLabel(
    String label, {
    Color textColor,
    String accessibilityLabel,
  }) {
    this._label = label;
    this._textColor = textColor;
    this._accessibilityLabel = accessibilityLabel;
  }

  @override
  String get type => "Label";
}
